assume/common/base.py

```diff
@@ -691,6 +691,43 @@ def calculate_ramp_charge(
         return power_charge
 
 
+class BaseDSTComponent:
```

Comment:
This behaves fundamentally different than the other BaseUnit types.
I would rather move this to dsm_load_shift.py?